src/foam/u2/crunch/CapabilityFeatureView.js

```diff
@@ -168,6 +169,21 @@ foam.CLASS({
           }
         });
       }
+    },
+    {
+      name: 'statusUpdate',
+      code: function() {
+        if ( this.cjStatus != this.CapabilityJunctionStatus.PENDING ) {
+          return;
+        };
+        this.crunchService.getJunction(null, this.data.id).then(ucj => {
+          if ( ucj && ucj.status === this.CapabilityJunctionStatus.GRANTED ) {
+            this.cjStatus = this.CapabilityJunctionStatus.GRANTED;
+          } else {
+            window.setTimeout(this.statusUpdate, 2000);
+          }
+        })
```

Comment:
Please add a catch incase the call fails.

Reply:
This doesn't do anything. If it runs in a .then(), then it is async, so throwing an exception after this code has already returned, can never be caught.

Reply:
@kgrgreer I think it works as .then().catch()

Reply:
The service method call can still fail

Reply:
Yes, it can fail, but what good with throwing an exception do? It will happen in the future after this method has returned and nobody can catch that exception.

Reply:
Then console.error the message